addons/account_edi/controller/main.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import http
+from odoo.http import request, content_disposition
+
+
+class EdiDocumentDownloadController(http.Controller):
+    @http.route('/web/edi_document/download', type='http', auth='user')
+    def download(self, ids=None):
+        ids = [int(id) for id in ids.split(',')]
+        res = request.env['account.move'].sudo().browse(ids)._create_zipped()
+        content = res.getvalue()
```

Comment:
```suggestion
        content = request.env['account.move'].sudo().browse(ids)._create_zipped()
```
With a change in _create_zipped

Reply:
Also not sure about the sudo here. as you are passing a list of id's in the URL, it means that anyone could print the edi documents of any invoices in the db if they're logged in, even if they shouldn't have access to them.